lib/smack/SmackInstGenerator.cpp

```diff
@@ -144,6 +144,38 @@ void SmackInstGenerator::processInstruction(llvm::Instruction& inst) {
   nextInst++;
 }
 
+void SmackInstGenerator::processIntrinsicCall(llvm::IntrinsicInst* ii) {
+  static const auto f16UpCast = [&ii, this] {
+    std::string funcName;
+    auto tyID = ii->getFunctionType()->getReturnType()->getTypeID();
+    if (tyID == llvm::Type::FloatTyID)
+      funcName = "__SMACK_convert_from_fp16_f32";
+    else if (tyID == llvm::Type::DoubleTyID)
+      funcName = "__SMACK_convert_from_fp16_f64";
+    return Stmt::call(funcName, {rep->expr(ii->getArgOperand(0))},
```

Comment:
What happens if neither `if` nor `else if` are executed?
I suspect we want to add here as a sanity check in future LLVM versions:
```
else
  llvm_unreachable("Unexpected return type of half-precision upcast intrinsic.");
```

Reply:
The else branch should not be executed because according to LLVM language manual, the resulting types should only be either float or double. I added the sanity check.